tools/file_packager.py

```diff
@@ -738,12 +777,51 @@ def was_seen(name):
   });
 '''
 
-ret += '''
+ret += '''%s
 })();
-'''
+''' % ('''
+  Module['removeRunDependency']('%(metadata_file)s');
+ }
+
+ var REMOTE_METADATA_NAME = typeof Module['locateFile'] === 'function' ?
+                            Module['locateFile']('%(metadata_file)s') :
+                            ((Module['filePackagePrefixURL'] || '') + '%(metadata_file)s');
+ var xhr = new XMLHttpRequest();
+ xhr.onreadystatechange = function() {
+  if (xhr.readyState == 4 && xhr.status == 200) {
+    loadPackage(JSON.parse(xhr.responseText));
+  }
+ }
+ xhr.open('GET', REMOTE_METADATA_NAME, true);
+ xhr.overrideMimeType('application/json');
+ xhr.send(null);
+
+ if (!Module['preRun']) Module['preRun'] = [];
+ Module["preRun"].push(function() {
+  Module['addRunDependency']('%(metadata_file)s');
+ });
+''' % {'metadata_file': os.path.basename(jsoutput + '.metadata')} if separate_metadata else '''
+ }
+ loadPackage();
+''')
+
 if force or len(data_files) > 0:
   if jsoutput == None:
     print ret
   else:
-    f = open(jsoutput, 'w')
-    f.write(ret)
+    if os.path.isfile(jsoutput):
```

Comment:
This part the of logic to check whether the file packager should overwrite the jsoutput file if it already exists from the prior invocation. It it exists then check if the old content is the same as the current content. Only overwrite when the content is different. Otherwise, leave the existing file untouched thus preserving its timestamp.